experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -38,17 +38,16 @@
 #define WS_SET_DUMMY_LINE_PERIOD		0x3A
 #define WS_SET_GATE_TIME			0x3B
 
-static unsigned width = 0;
-static unsigned height = 0;
-static unsigned bpp = 0;
-
 struct ws_eink_fb_par {
 	struct spi_device *spi;
 	struct fb_info *info;
 	u8 *ssbuf;
 	int rst;
 	int dc;
 	int busy;
+	int width;
```

Comment:
Rather than adding a width, height and bpp member here, I think it would be better to add a `const struct waveshare_eink_device_properties *properties` member.  Then you could do `par->properties->width` if you wanted to look at the width.